pkg/services/ngalert/sender/router.go

```diff
@@ -230,7 +258,40 @@ func (d *AlertsRouter) alertmanagersFromDatasources(orgID int64) ([]string, erro
 				"error", err)
 			continue
 		}
-		alertmanagers = append(alertmanagers, amURL)
+		m, err := ds.JsonData.Map()
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch headers for datasource: %w", err)
+		}
+
+		const headerName = "httpHeaderName"
```

Comment:
It's kind of a bonkers that such a sensitive thing is not export as a constant directly from the datasource package - I'd add these for both `httpHeaderValue` and `httpHeaderName ` - and replace the string usage with constants.